client/src/main/java/io/pravega/client/netty/impl/RawClient.java

```diff
@@ -80,6 +92,12 @@ public void authTokenCheckFailed(WireCommands.AuthTokenCheckFailed authTokenChec
         }
     }
 
+    public RawClient(PravegaNodeUri uri, ConnectionFactory connectionFactory) {
+        this.segmentId = null;
```

Comment:
If segmentId is going to be null sometimes, you might as well just remove the field. It's only used in the log message when closing the connection anyway.

Reply:
how about use a separate PR to remove segmentId from RawClient and all the callers and their test cases, so we can go ahead merge this PR first. thank you.